grails-app/services/org/pih/warehouse/core/DashboardService.groovy

```diff
@@ -163,6 +163,14 @@ class DashboardService {
             expiredStock = expiredStock.findAll { item -> item?.product?.category == category }
         }
 
+        if (startDate) {
+            expiredStock = expiredStock.findAll { item -> item?.expirationDate >= startDate }
+        }
+
+        if (endDate) {
+            expiredStock = expiredStock.findAll { item -> item?.expirationDate <= endDate }
+        }
```

Comment:
this feels odd to me - this should be probably filtered out at the query level, but I can see that you've followed the convention, so I don't have an opinion if it should be handled now or if we should treat it as a tech debt.